eZ/Bundle/EzPublishElasticsearchBundle/DependencyInjection/EzPublishElasticsearchExtension.php

```diff
@@ -35,5 +33,79 @@ public function load( array $configs, ContainerBuilder $container )
         );
         $loader->load( 'indexable_fieldtypes.yml' );
         $loader->load( "search_engines/elasticsearch.yml" );
+
+        $loader = new YamlFileLoader(
+            $container,
+            new FileLocator( __DIR__ . '/../Resources/config' )
+        );
+        $loader->load( 'services.yml' );
+
+        $this->processConnectionConfiguration( $container, $config );
+    }
+
+    /**
+     * Processes connection configuration by flattening connection parameters
+     * and setting them to the container as parameters.
+     *
+     * @param \Symfony\Component\DependencyInjection\ContainerBuilder $container
+     * @param array $config
+     */
+    protected function processConnectionConfiguration( ContainerBuilder $container, $config )
+    {
+        $alias = $this->getAlias();
+
+        if ( isset( $config["default_connection"] ) )
+        {
+            $container->setParameter(
+                "{$alias}.default_connection",
+                $config["default_connection"]
+            );
+        }
+
+        foreach ( $config["connections"] as $name => $params )
+        {
+            $flattenedParams = $this->flattenParams(
+                $params,
+                "{$alias}.connection.{$name}"
+            );
+
+            foreach ( $flattenedParams as $key => $value )
+            {
+                $container->setParameter( $key, $value );
+            }
+        }
+    }
+
+    /**
+     * Flattens nested array structure into a single level key-value array, concatenating
+     * keys through levels and keeping values.
+     *
+     * @param array $nestedParams
+     * @param string $prefix
+     *
+     * @return array
+     */
+    protected function flattenParams( $nestedParams, $prefix )
```

Comment:
I don't see how this method is needed. Container can manage deep hashes for one key